option optimize_for = SPEED;
package ProtoGameClient;

enum REQUEST_MSG
{
	HEART = 1000;
	FLAGS = 1001;
	LOGIN = 1002;
	LIST_GAME = 1003;
	CREATE_GAME = 1004;
	DESTROY_GAME = 1005;
	ENTER_GAME = 1006;
	EXIT_GAME = 1007;
	SEND_TO_PLAYER = 1008;
	SEND_TO_PLAYER_ALL = 1009;
}

message Heart
{
	required int32 timestamp = 1;
}

message Flags
{
}

message Login
{
	required uint32 guid = 1;
	required int32 version = 2;
}

message ListGame
{
}

message CreateGame
{
	required string password = 1;
	required int32 map = 2;
	required int32 mode = 3;
	required int32 maxPlayers = 4;
	required float evaluation = 5;
}

message DestroyGame
{
}

message EnterGame
{
	required uint32 gameid = 1;
	required string password = 2;
}

message ExitGame
{
}

//message ModifyGamePassword
//{
//	required string password = 1;
//}

message SendToPlayer
{
	required uint32 guid = 1;
	required int32 size = 2;
	required bytes data = 3;
}

message SendToPlayerAll
{
	required uint32 filter = 1;
	required int32 size = 2;
	required bytes data = 3;
}